Library/HUMSlider.m

```diff
@@ -109,25 +164,195 @@ - (void)nukeOldTicks
     [self layoutIfNeeded];
 }
 
+- (void)refreshView {
+    [self setupTicks];
+}
+
 - (void)setupTicks
 {
+    if ([self areCustomTicksSetupAndNonNull]) {
+        [self cleanupAfterEvenlySpacedTicks];
+        [self setupCustomTickViews];
+        [self setupTicksAutoLayoutCustomWidths];
+        if (!_enableTicksTransparencyOnIdle) {
+            [self animateAllTicksInCustomWidths:YES];
+        }
+    }
+    else {
+        [self cleanupAfterCustomSpacedTicks];
+        [self setupSpacedTickViews];
+        [self setupTicksAutolayout];
+        if (!_enableTicksTransparencyOnIdle) {
+            [self animateAllTicksIn:YES];
+        }
+    }
+    
+
+}
+
+- (void)cleanupAfterEvenlySpacedTicks {
+    self.tickViews = [NSArray new]; // Make sure these get re-initialized
+    [self clearLayoutConstraintList:_leftTickRightConstraints];
+    [self clearLayoutConstraintList:_rightTickLeftConstraints];
+    [self clearLayoutConstraintList:_allTickBottomConstraints];
+}
+
+- (void)cleanupAfterCustomSpacedTicks {
+    self.tickViews = [NSArray new]; // Make sure these get re-initialized
+    [self clearLayoutConstraintList:_tickLeftConstraints];
+    [self clearLayoutConstraintList:_tickRightConstraints];
+    [self clearLayoutConstraintList:_allTickBottomConstraints];
+}
+
+// Disassociate all of the NSLayoutConstraints in the list from the parent view.
+- (void)clearLayoutConstraintList:(NSArray*)list {
+    for (NSLayoutConstraint *constraint in list) {
+        [self removeConstraint:constraint];
+    }
+}
+
+- (void)setupSpacedTickViews {
+    [self createAndAddBlankTickViewsWithCount:(int)self.sectionCount];
+}
+
+- (void)setupCustomTickViews {
+    NSLog(@"Setting up Custom tick views for %d custom ticks.", (int)[_ticks count]);
+    [self createAndAddBlankTickViewsWithCount:(int)[_ticks count]];
+}
+
+- (bool)areCustomTicksSetupAndNonNull {
+    return (_customTicksEnabled && _ticks && [_ticks count] > 0);
```

Comment:
Generally outside of their initializers, in obj-c we try not to access the underlying variables and use `self.` to access properties. This ensures custom getters and setters are fired every time.